### PR TITLE
Broker on client windows allows only 1 user to create sessions.

### LINKED ISSUE #2330: Broker should not allow multiple clients on plain Windows
Windows Server should be required for shared environment.

### PR DESCRIPTION
**Pull request by karthiknadig**
Fixes #2330

### CODE REVIEW

File: src/Host/Broker/Impl/NativeMethods.cs [hunk 1]
```diff
@@ -52,5 +94,19 @@ internal static unsafe class NativeMethods {
...
+        static extern ulong VerSetConditionMask(ulong dwlConditionMask, uint dwTypeBitMask, byte dwConditionMask);
+
+        [DllImport("kernel32.dll")]
+        static extern bool VerifyVersionInfo([In] ref OSVERSIONINFOEX lpVersionInfo, uint dwTypeMask, ulong dwlConditionMask);
+
+        public static bool IsWindowsServer() {
+            OSVERSIONINFOEX osVersionInfo = default(OSVERSIONINFOEX);
+            osVersionInfo.dwOSVersionInfoSize = (uint)Marshal.SizeOf(osVersionInfo);
+            osVersionInfo.wProductType = VER_NT_WORKSTATION;
+            ulong mask = VerSetConditionMask(0, VER_PRODUCT_TYPE, VER_EQUAL);
```
Comment (MikhailArkhipov) [inline src/Host/Broker/Impl/NativeMethods.cs]: https://msdn.microsoft.com/en-us/library/bb773795(v=VS.85).aspx may be simpler to use

File: src/Host/Broker/Impl/Resources.resx [hunk 2]
```diff
@@ -237,4 +237,7 @@
   <data name="Trace_UserProfileDirectory" xml:space="preserve">
     <value>User {0} profile directory: {1}</value>
   </data>
+  <data name="Exception_MaxAllowedUsers" xml:space="preserve">
+    <value>Broker has reached maximum allowed concurrent user limit.</value>
```
Comment (MikhailArkhipov) [inline src/Host/Broker/Impl/Resources.resx]: If this is user-visible I'd split it into 2 specific messages. `Windows Workstation only allows a singe user session' and 'R Remote Services computer has reached maximum number of concurrent users'

Comment (karthiknadig) [inline src/Host/Broker/Impl/Resources.resx]: Where should the second message come from? The first one comes from the session manager in broker.

Comment (MikhailArkhipov) [inline src/Host/Broker/Impl/Resources.resx]: I see that we throw ArgumentOutOfRange exception. If you try it, does it get propagated to the client via BrokerApiError? If so, then there should be error code that will map to the appropriate message in REPL. See `BrokerApiErrorException`

Comment (karthiknadig) [inline src/Host/Broker/Impl/Resources.resx]: I have made the change:
image

**Comment by karthiknadig** [comment]
Ping.

File: src/Host/Broker/Impl/Resources.resx [hunk 3]
```diff
@@ -237,4 +237,7 @@
   <data name="Trace_UserProfileDirectory" xml:space="preserve">
     <value>User {0} profile directory: {1}</value>
   </data>
+  <data name="Exception_MaxAllowedUsers" xml:space="preserve">
+    <value>Windows Workstation only allows a singe user session.</value>
```
Comment (huguesv) [inline src/Host/Broker/Impl/Resources.resx]: singe means monkey in french.

File: src/Host/Client/Impl/Resources.resx [hunk 4]
```diff
@@ -188,4 +188,10 @@ If you using a test Remote R Server with a self-signed certificate and are certa
     <value>Previous R session terminated unexpectedly, and its global workspace is currently being saved.
 Do you want to abort this operation and start current session immidiately?</value>
   </data>
+  <data name="Error_MaxUsersAllowed" xml:space="preserve">
+    <value>Windows Workstation only allows a singe user session.</value>
```
Comment (huguesv) [inline src/Host/Client/Impl/Resources.resx]: same typo

**Pull request merged**